internal/pkg/provisioner/storageclass.go

```diff
@@ -99,29 +99,33 @@ func StorageClassReservedParams(sc *storagev1.StorageClass) map[string]string {
 }
 
 // ValidateOrSetStorageClassUID returns true if the StorageClass annotation on
-// the PVC object matches the uid passed in.
+// the PVC object matches the uid passed in or the StorageClass is younger than PVC.
```

Comment:
This too can be improved based on the change. `matches the uid of the passed in SC...`.
And should it return true if the SC is older than PVC? As per the comments below, younger SC isn't safe and returns false.

Reply:
Sorry mate, i don't understand the second part. It returns true if the SC is older and updates was ok: https://github.com/storageos/api-manager/blob/5dd47fbacd5ece908761c3f6612226df916ed7d5/internal/pkg/provisioner/storageclass.go#L130

Could you please explain?

Reply:
Yes, so it should return true if SC is older. The above comment states that it is returning true if SC is younger.

Referring to https://github.com/storageos/api-manager/pull/63/files#diff-9c56144171130b386f3a5dab17f4c3af7fc963fdda728ce78287cb89ab31d1dcR113-R116

> 	// If StorageClass is younger than the PVC itself, the synchronization isn't safe.